### ISSUE TITLE
Better ux for when there's no backuplocation

### ISSUE THREAD

**Issue by carlisia** [heptio/ark#979]
I purposefully started the ark server w/o setting a backup location via the cli or deploying the `backupstoragelocation.yaml` file.

In the end, the backup sync kept retrying for a really long time.

I wonder if the warning at the top should be an error, if the message should be stronger, of else there is a better way to avoid the user being stuck in this scenario.

![image](https://user-images.githubusercontent.com/16508/47325521-0f0f7d80-d619-11e8-93f2-bd45ed6010f4.png)

**Comment by carlisia** [comment]
For reference: https://github.com/heptio/ark/issues/898#issuecomment-429047146

**Comment by carlisia** [comment]
Question: is it expected that Ark be used only for PV backup?

**Comment by ncdc** [comment]
> In the end, the backup sync kept retrying for a really long time.

This is acceptable and a reasonable design, imho. It should keep trying until it has work to do.

> I wonder if the warning at the top should be an error, if the message should be stronger, of else there is a better way to avoid the user being stuck in this scenario.

What if we added an `ark status` command that could give you an overview of the server's status? Things like "waiting on BSL", "waiting on VSL", etc?

**Comment by ncdc** [comment]
> Question: is it expected that Ark be used only for PV backup?

Are you asking if it should handle volumes that are directly referenced by the pod (e.g. `spec.volumes[i].gcePersistentDisk`)? If so, that's #408

**Comment by carlisia** [comment]
No, I'm asking something different. Currently we have backup storage and volume storage. Can we bootup the ark server w/o setting up/deploying configuration for the backup storage if we only want to do pv storage?

**Comment by carlisia** [comment]
> What if we added an ark status command that could give you an overview of the server's status? Things like "waiting on BSL", "waiting on VSL", etc?

Yes, I'd like that. Also, could we change the message from `Checking for backup storage` to `Waiting for backup storage`? I think the later gives an idea that there's a user action that needs to be taken.

**Comment by ncdc** [comment]
> No, I'm asking something different. Currently we have backup storage and volume storage. Can we bootup the ark server w/o setting up/deploying configuration for the backup storage if we only want to do pv storage?
>
> I know the answer to that ^ is no, but should we?

That is along the lines of #504. I'm interested in exploring this, but I'm not sure how/where we'd record the information (volume snapshot info).

**Comment by ncdc** [comment]
It depends on how we implement it. I could see 3 options:

1. The `ark` server keeps a new status-y CR up to date and `ark status` would retrieve the CR's contents. This would potentially report false positives, e.g. in the event the `ark` server is offline.
1. `ark status` creates and sends a new `StatusRequest` CR, and the `ark` server has a controller that updates it. The client waits/watches for it to be processed.
1. The `ark` server exposes an HTTP endpoint that the client communicates with.

Of these, I'd probably vote for 2.

**Comment by nrb** [comment]
I like 2, too. Seems like a lot less plumbing for users to have to wrangle.

**Comment by carlisia** [comment]
I like 1 better. I'm thinking when I want to check on a status it's bc something is happening/not happening and I want to get a quick response. Sound that there'd be a latency with option 2. For option 1, could the `ark status` not ping and check if the server is online? If not online maybe add a disclaimer saying so and providing last known status?

**Comment by carlisia** [comment]
Adding this for future reference: we might consider adding to the `ark status` a field that indicates what the ark server looks for as far as default locations. Currently, if there's more than 1 location we can't tell which one is the default:

![image](https://user-images.githubusercontent.com/16508/47448027-d46e2800-d773-11e8-85b0-df393d3bd7cc.png)

**Comment by nrb** [comment]
Given this wasn't included in #1052, I'm removing it from the v0.10.0 milestone

**Comment by rosskukulinski** [comment]
> > In the end, the backup sync kept retrying for a really long time.
> 
> This is acceptable and a reasonable design, imho. It should keep trying until it has work to do.

@ncdc If Ark doesn't have a functioning backup storage location, the process should eventually exit.  If the Ark pod is running then I would expect that it's functional.  Alternatively, we could leverage a liveness probe to help let users know that Ark is not in a happy state.

**Comment by ncdc** [comment]
I'm afraid that if we have the server crashloop because something such as the default BSL is missing, we're going to get bug reports that the server is crashlooping. Do you agree or disagree?

**Comment by rosskukulinski** [comment]
@ncdc ahh! I didn't realize that this is related to a missing _default_.  Agree crashlooping for that is bad.

**Comment by carlisia** [comment]
This is different, but related: https://github.com/vmware-tanzu/velero/issues/675.

**Comment by eleanor-millman** [comment]
Looks like @codegold79 's Design doc might address this: https://github.com/vmware-tanzu/velero/pull/4270/files

Frankie, can you please take a look at this issue and mark it as a dupe of whatever other issue you are working off of if this is the case? Thanks!
